zebra-chain/src/primitives/zcash_history.rs

```diff
@@ -140,15 +140,15 @@ impl<V: Version> Tree<V> {
     ///  (ignored for V1 trees).
     #[allow(clippy::unwrap_in_result)]
     pub fn new_from_block(
-        network: Network,
+        network: &Network,
         block: Arc<Block>,
         sapling_root: &sapling::tree::Root,
         orchard_root: &orchard::tree::Root,
     ) -> Result<(Self, Entry), io::Error> {
         let height = block
             .coinbase_height()
             .expect("block must have coinbase height during contextual verification");
-        let network_upgrade = NetworkUpgrade::current(network, height);
+        let network_upgrade = NetworkUpgrade::current(&network, height);
```

Comment:
Looks like a double borrow.

```suggestion
        let network_upgrade = NetworkUpgrade::current(network, height);
```

Reply:
change applied